mindsdb/integrations/handlers/bigcommerce_handler/bigcommerce_tables.py

```diff
@@ -0,0 +1,1420 @@
+from typing import List
+from decimal import Decimal
+
+import pandas as pd
+
+from mindsdb.integrations.handlers.bigcommerce_handler.bigcommerce_api_client import BigCommerceAPIClient
+from mindsdb.integrations.utilities.sql_utils import FilterCondition, FilterOperator, SortColumn
+from mindsdb.integrations.libs.api_handler import MetaAPIResource
+from mindsdb.utilities import log
+
+
+logger = log.getLogger(__name__)
+
+
+def _make_filter(conditions: list[FilterCondition] | None, op_map: dict) -> dict:
+    """Creates a filter dictionary, that can be used in the BigCommerce API.
+
+    Args:
+        conditions (list[FilterCondition]): The list of parsed filter conditions.
+        op_map (dict): The mapping of filter operators to API parameters.
+
+    Returns:
+        dict: The filter dictionary.
+    """
+    filter = {}
+    if conditions is None:
+        return filter
+    for condition in conditions:
+        simple_op = op_map.get((condition.column, condition.op))
+        if simple_op:
+            value = condition.value
+            if isinstance(value, list):
+                value = ",".join(map(str, value))
+            filter[simple_op] = value
+            condition.applied = True
+    return filter
+
+
+def _make_df(result: list[dict], table: MetaAPIResource):
+    """Converts a list of dictionaries to a pandas DataFrame.
+    If the list is empty, an empty DataFrame is returned with the columns from the table.
+
+    Args:
+        result (list[dict]): The list of dictionaries to convert.
+        table (MetaAPIResource): The table class.
+
+    Returns:
+        pd.DataFrame: The resulting DataFrame.
+    """
+    if len(result) == 0:
+        result = pd.DataFrame([], columns=table.get_columns())
+    else:
+        result = pd.DataFrame(result)
+    return result
+
+
+def _make_sort_condition_v3(sort: list[SortColumn], sortable_columns: list[str]):
+    """Creates a sort condition for the BigCommerce API v3.
+
+    Args:
+        sort (list[SortColumn]): The list of parsed sort columns.
+        sortable_columns (list[str]): The list of sortable columns.
+
+    Returns:
+        dict: The sort condition, that can be used in the BigCommerce API v3.
+    """
+    sort_condition = None
+    if isinstance(sort, list) and len(sort) == 1 and sort[0].column in sortable_columns:
```

Comment:
maybe we can remove condition `len(sort) == 1` and use the first column that in sortable_columns?